packages/schema-derive/src/query_responses.rs

```diff
@@ -40,6 +67,30 @@ fn parse_query(v: Variant) -> (String, Expr) {
     )
 }
 
+/// Extract the nested query  -> response mapping out of an enum variant.
+fn parse_subquery(v: Variant) -> Expr {
+    let submsg = match v.fields {
+        syn::Fields::Named(_) => panic!("nested query response can only contain tuple structs"),
+        syn::Fields::Unnamed(fields) => {
+            if fields.unnamed.len() != 1 {
+                panic!("invalid number of subquery parameters");
+            }
+
+            fields.unnamed[0].ty.clone()
+        }
+        syn::Fields::Unit => panic!("a unit variant is not a valid subquery"),
+    };
+    parse_quote!(<#submsg as ::cosmwasm_schema::QueryResponses>::response_schemas_impl())
+}
+
+/// Checks whether the input has the given #[query_responses($attr))] attribute
```

Comment:
Ah, this is more flexible than the doc comment suggests. Outdated comment?

Reply:
Good catch